spec/support/shared_examples/return_to_review_concern.rb

```diff
@@ -0,0 +1,34 @@
+require "rails_helper"
+
+shared_examples :return_to_review_concern do
+  # requires form_params to be set
+  describe "#next_path" do
+    context "with return_to_review param set" do
+      it "navigates to the state review screen" do
+        post :update, params: form_params.merge({return_to_review: "y"})
+
+        case form_params[:us_state]
+        when "az"
+          expect(response).to redirect_to(controller: "az_review", action: :edit, us_state: "az")
+        when "ny"
+          expect(response).to redirect_to(controller: "ny_review", action: :edit, us_state: "ny")
+        end
+      end
+    end
+
+    context "without return_to_review_param set" do
+      it "navigates to the next page in the flow" do
+        post :update, params: form_params
+        controllers = []
+        case form_params[:us_state]
+        when "az"
+          controllers = Navigation::StateFileAzQuestionNavigation::FLOW.to_a
+        when "ny"
+          controllers = Navigation::StateFileNyQuestionNavigation::FLOW.to_a
+        end
+        next_controller = controllers[controllers.index(described_class) + 1]
```

Comment:
this obvs has some edge cases where it could break, but i think it's safe to assume that we should be testing controllers that have a valid next controller in this context.